### ISSUE TITLE
Anomaly between EF LocalDB and SqLite using same classes

### ISSUE THREAD

**Issue by jhalbrecht** [dotnet/efcore#7668]
I reached out on Twitter to Rowan Miller @rowanmiller  who asked me to submit this issue.

In the SqLite version when a new Blog that contains some Posts are added to the current DbContext (BloggingContext) the relationship exists between the Blogs and it's Posts can be seen. Once that specific context is released, after a SaveChanges() the relationship is gone although the individual Blogs and Posts are still in the SqLite database. With the LoclDB version the relationship is maintained between closing and reopening the context.

I placed a simple distilled console app demonstrating the problem in this repository [DotNetEfCoreConsoleBlogging](https://github.com/jhalbrecht/DotNetCoreEfBlogging.Console) The Blog and Post classes were taken directly from Microsoft examples such as [Entity Framework Core](https://docs.microsoft.com/en-us/ef/core/index).

### Steps to reproduce
I've set a Debugger.Break(); in stratgetic points in the code. Examine the current DbContext before and after it is released. Relationship of `List<Post>` in Blog exists until the current DbContext is released.

Microsoft.EntityFrameworkCore 1.1.0
Microsoft.EntityFrameworkCore.Sqlite 1.1.0
Microsoft.EntityFrameworkCore.Tools 1.1.0-preview4-final (or v1.1.0-msbuild3-final)
Microsoft.EntityFrameworkCore.Design 1.1.0
Visual Studio 2017 RC 4
Windows 10 Pro Version	10.0.14393 Build 14393

    public class Blog
    {
        public Blog()
        {
            Posts = new List<Post>();
        }
        public int BlogId { get; set; }
        public string Name { get; set; }
        //public virtual List<Post> Posts { get; set; }
        public List<Post> Posts { get; set; }
        //public virtual ICollection<Post> Posts { get; set; }
    }

    public class Post
    {
        public int PostId { get; set; }
        public string Title { get; set; }
        public string Content { get; set; }
        public int BlogId { get; set; }
        // public virtual Blog Blog { get; set; }
        public Blog Blog { get; set; }
    }

**Comment by ajcvickers** [comment]
@jhalbrecht I am unable to reproduce this, meaning that I see the same behavior on SQL Server and SQLite. In both cases, the initial call to load Blogs after seeding does not also load the Posts, which is why they are not present in the loaded graph. Use an Include to also load the posts:
```C#
var blogs = db.Blogs.Include(b => b.Posts).ToList();
```
In CheckEmOut, the queries are not being evaluated because there is no code like a `ToList` or similar to cause the queries to be evaluated. If I add ToList to both the Blogs and the Posts query, then the result it everything loaded with relationships correctly fixed up.

In summary, I don't see anything that isn't by-design here, and I see the same results on both SQL Server and SQLite. Could you try again and see if you still get different results with the repro code? If not, maybe it needs to be tweaked to see what you were previously seeing.

**Comment by jhalbrecht** [comment]
### A bit more background
The sample code in this condensed console example is distilled from my repository [EfBlogging](https://github.com/jhalbrecht/EfBlogging) which contains a WPF with LocalDB  and a UWP with SqLite example utilizing the exact same Blog and Post classes. I filed [an issue](https://github.com/jhalbrecht/EfBlogging/issues/1) on my own repository with extensive screenshots illustrating the anomaly between the LocalDB and the SqLite versions. The WPF / LocalDB version retains the relationship after closing and reopening the DbContext the SqLite version does not retain the relationship between closing and reopening the DbContext. In each of the ViewModels for the WPF and the UWP I initialize the same way `Blogs = new ObservableCollection<Blog>(bloggingContext.Blogs);`

### var blogs = db.Blogs.Include(b => b.Posts).ToList();
I modified my Console app adding your `var blogs = db.Blogs.Include(b => b.Posts).ToList();` surrounded by a Debugger.Break(); To my amazement merely querying with `var blogs = db.Blogs.Include(b => b.Posts).ToList();`  (not modifying or adding) the bloggingContextAfter changed such that the `List<Post>` property of the blogs was now enumerated. Can or should a DbContext change with just a query? 

        private static void CheckEmOut()
        {
            using (var bloggingContextAfter = new BloggingContext())
            {
                var blogs = bloggingContextAfter.Blogs;
                // there is no List<Post> in the bloggingContextAfter.blogs, well it does have a count=0
                Debugger.Break(); 
                var ajblogs = bloggingContextAfter.Blogs.Include(b => b.Posts).ToList();
                // Now the bloggingContextAfter.blogs does contain a List<Post> with a count > 0
                Debugger.Break(); 
                Logger.Log($"CheckEmOut count posts in first Blog => {blogs.FirstOrDefault().Posts.Count} <=");
                var posts = bloggingContextAfter.Posts;
                Logger.Log("Check out the Blogs in the bloggingContextAfter DbContext. Posts count = 0");
                Debugger.Break();
            }
        }

Note that _the wpf / LocalDB version using .NET EntityFramework 6.1.3 does not require this_. Perhaps I'm comparing apples to oranges? The ef in .netCore != .net Framework 4.6. However should I expect similar behavior?

**Comment by ajcvickers** [comment]
@jhalbrecht "Can or should a DbContext change with just a query?" When a new DbContext instance is created it will initially not be tracking any entities. The context is typically populated with entities by querying them or attaching them. It might be worth reading through some of the documentation here to understand the concepts involved: https://docs.microsoft.com/en-us/ef/core/

The EF6 behavior is different because EF6 can do lazy loading of relationships. EF Core doesn't support lazy loading, so the code must either eagerly load relationships (using Include) or explicitly load them (using, for example, `context.Entry(foo).Collection(e => e.Posts).Load();` Whether or not lazy loading is a good thing is hotly debated. See #3797.
